base/compiler/validation.jl

```diff
@@ -62,7 +62,7 @@ end
 InvalidCodeError(kind::AbstractString) = InvalidCodeError(kind, nothing)
 
 function validate_code_in_debug_mode(linfo::MethodInstance, src::CodeInfo, kind::String)
-    if JLOptions().debug_level == 2
+    if ccall(:jl_is_debugbuild, Cint, ()) == 1
```

Comment:
Maybe we should do this on an asserts build too.

Reply:
Though I guess we could err on the side of safety and verify it here as well

Reply:
> this one is looking at the debug info itself

Is it? It just calls `validate_code(linfo, src)`, so that seems like regular validation.

Reply:
I got confused because I read
> let's validate linfo 
And misunderstood that as debuginfo, but it's just an MI. So yeah, I think this needs to be on as well

Reply:
We should be careful not to introduce anything too expensive though; PkgEval runs with assertions by default, and we're already at 16 hours per (full) evaluation.

Reply:
Can we get a rough estimate by running it on this PR?

Reply:
I'll compare the timings of compiling corecompiler.

Reply:
Looks like a 5% slowdown... so not nothing. We can try it out on PkgEval and possibly revert this part of the change.

Reply:
What is the slowdown of just the other verify_ir calls

Reply:
Only from the extra `validate_code_on_debug_build` calls.

Reply:
Lets do the non `validate_code_on_debug_build` for now, since it probably will catch quite a few things, and in the future we can look into adding these as well.

Reply:
I'll try a run first, just to see how much this would uncover.